### ISSUE TITLE
On jsx-no-bind.md, Put the description for Function Components on the Rule Details section

### ISSUE THREAD

**Issue by arthurRuf** [jsx-eslint/eslint-plugin-react#3168]
On https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md page, the description of what is the correct code for Function Components is displayed on the end of the page, which made some teammates miss it.

The current description on “Rule Details” works only for Class Components and can be misleading for Function Components cases.

As Function Components are not an exception in React, it would be better to display the “correct code” for this type of component on the top of the page.

TL;DR;
Take a look on the image :) 
![image](https://user-images.githubusercontent.com/8228665/148247963-d45476aa-4639-4016-9da5-bb29b822c809.png)
Why? Because it's on the end of the page and people miss it.

**Comment by ljharb** [comment]
We're pretty consistent about "rule options" coming first; and there's a Table of Contents button in the upper left hand side of every markdown file on github that would show you the "react hooks" section. I'm not sure there's a better fix here besides "read more carefully".
